stdlib/public/core/StringIndex.swift

```diff
@@ -38,8 +39,11 @@ extension String.Index : Comparable {
 extension String.Index {
   internal typealias _Self = String.Index
   
-  public init(encodedOffset o: Int) {
-    _compoundOffset = UInt64(o << _Self._strideBits)
+  /// Creates a new index at the specified UTF-16 offset.
```

Comment:
FYI, I aim to decouple this stuff from UTF-16 eventually.  That is really meant to be a code unit offset in some arbitrary encoding.  I'd have labeled it `utf16Offset` otherwise.